rclcpp/include/rclcpp/node.hpp

```diff
@@ -153,6 +153,11 @@ class Node : public std::enable_shared_from_this<Node>
   const std::vector<rclcpp::CallbackGroup::WeakPtr> &
   get_callback_groups() const;
 
+  /// Iterate over the callback groups in the node, calling func on each valid one.
+  RCLCPP_PUBLIC
+  void
+  callback_groups_for_each(const node_interfaces::NodeBaseInterface::CallbackGroupCallback & func);
```

Comment:
nitpick: the name sounds weird to me, what about `for_each_callback_group()`? That's _closer_ to being a verb/action than the current name. The current name makes me think like the C api's where it is a `<class_name_method_name>` pattern and implies `callback_groups` is a class, but it isn't.

Another, more complicated solution would be to create a new container like `CallbackGroups` or something which has a `for_each()` method on it.

That all being said, it's just bikeshedding.

Reply:
Also, either the name of the method, or the documentation should probably mention the purpose of this method, i.e. that it provides thread-safety by atomically iterating over the callbacks groups (for the whole iteration, as opposed to locking in each loop) and synchronizing with adding and/or deleting callback groups.

Reply:
> nitpick: the name sounds weird to me, what about `for_each_callback_group()`

Ivan also mentioned this, so I'll take that as 2 votes for `for_each_callback_group()`.  I'll change to that instead.

Reply:
> Also, either the name of the method, or the documentation should probably mention the purpose of this method, i.e. that it provides thread-safety by atomically iterating over the callbacks groups (for the whole iteration, as opposed to locking in each loop) and synchronizing with adding and/or deleting callback groups.

Yeah, I'll mention that when I write the documentation.